libraries/persistent_store/fuzz/src/stats.rs

```diff
@@ -0,0 +1,187 @@
+
+    /// The number of times a remove update was applied.
+    RemoveCount,
+
+    /// The number of times a store operation was interrupted.
+    InterruptionCount,
+}
+
+/// Statistics about multiple fuzzing runs.
+#[derive(Default)]
+pub struct Stats {
+    /// Maps each statistics to its histogram.
+    stats: HashMap<StatKey, Histogram>,
+}
+
+impl Stats {
+    /// Adds a measure for a statistics.
+    pub fn add(&mut self, key: StatKey, value: usize) {
+        self.stats.entry(key).or_default().add(value);
+    }
+
+    /// Merges another statistics into this one.
+    pub fn merge(&mut self, other: &Stats) {
+        for (&key, other) in &other.stats {
+            self.stats.entry(key).or_default().merge(other);
+        }
+    }
+
+    /// Returns the count of a bucket for a given key.
+    pub fn get_count(&self, key: StatKey, bucket: usize) -> Option<usize> {
+        self.stats.get(&key).and_then(|h| h.get(bucket))
+    }
+
+    /// Returns the bit-width of one past the highest non-empty bucket.
+    ///
+    /// In other words, all non-empty buckets of the histogram have a bit-width smaller than the
+    /// returned width.
+    fn width_lim(&self) -> usize {
+        self.stats
+            .values()
+            .map(|h| h.width_lim())
+            .max()
+            .unwrap_or(0)
+    }
+}
+
+impl std::fmt::Display for Stats {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> Result<(), std::fmt::Error> {
+        let mut matrix: Vec<Vec<String>> = Vec::new();
+        let bits = self.width_lim();
+
+        let mut header = Vec::new();
+        header.push(String::new());
+        for width in 0..bits {
+            header.push(format!(" {}", bucket_from_width(width)));
+        }
+        header.push(" count".into());
+        matrix.push(header);
+
+        for key in StatKey::iter() {
+            let mut row = Vec::new();
+            row.push(format!("{}:", key));
+            for width in 0..bits {
+                row.push(match self.get_count(key, bucket_from_width(width)) {
+                    None => String::new(),
+                    Some(x) => format!(" {}", x),
+                });
+            }
+            let count = self.stats.get(&key).map_or(0, |h| h.count());
+            row.push(format!(" {}", count));
+            matrix.push(row);
+        }
+
+        write_matrix(f, matrix)
+    }
+}
+
+/// Prints a string aligned to the right for a given width.
+fn align(f: &mut std::fmt::Formatter, x: &str, n: usize) -> Result<(), std::fmt::Error> {
+    for _ in 0..n.saturating_sub(x.len()) {
+        write!(f, " ")?;
+    }
+    write!(f, "{}", x)
+}
+
+/// Prints a matrix with columns of minimal width to fit all elements.
+fn write_matrix(
+    f: &mut std::fmt::Formatter,
+    mut m: Vec<Vec<String>>,
+) -> Result<(), std::fmt::Error> {
+    if m.is_empty() {
+        return Ok(());
+    }
+    let num_cols = m.iter().map(|r| r.len()).max().unwrap();
+    let mut col_len = vec![0; num_cols];
```

Comment:
Array length is a static property in Rust (same as non-VLA arrays in C and `array` in C++ if I'm not mistaken). Usually when I like to emphasize that a dynamic-length array is an array and not a vector, I do `vec![x; n].into_boxed_slice()` (i.e. no `push`/`pop`). But here this is a local variable (i.e. not cross-functional) and the function is short (understanding how it's used is a matter of seconds), so I don't think it's useful. I can change in the next PR if you think otherwise.

Reply:
Ah oops I missed the `num_cols` problem. You're good.